ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift

```diff
@@ -43,13 +54,33 @@ final class SidebarViewController: UITableViewController {
 
         switch switchType {
         case .timeline:
-            rootViewController?.navigationDrawerController?.toggleLeftView()
+            if let _ = UIApplication.topViewController() as? FilterViewController {
+                navigationDrawerController?.toggleLeftView()
+            } else {
+                let vc = FilterViewController()
+                let nvc = NavigationController(rootViewController: vc)
+                let root = NavigationDrawerController(rootViewController: nvc, leftViewController: SidebarViewController.instantiate(rootViewController: nvc, dependency: .init(switchType: switchType)))
+                navigationDrawerController?.transition(to: root, completion: { _ in
+                    self.navigationDrawerController?.toggleLeftView()
+                    })
+            }
+
         case .about:
             break
         case .info:
             break
         case .map:
-            break
+            if let _ = UIApplication.topViewController() as? FloorMapViewController {
+                navigationDrawerController?.toggleLeftView()
+            } else {
+                let vc = FloorMapViewController.instantiate()
+                let nvc = NavigationController(rootViewController: vc)
+                let root = NavigationDrawerController(rootViewController: nvc, leftViewController: SidebarViewController.instantiate(rootViewController: nvc, dependency: .init(switchType: switchType)))
+                navigationDrawerController?.transition(to: root, completion: { _ in
+                    self.navigationDrawerController?.toggleLeftView()
+               })
+            }
```

Comment:
In this part, we should match implement way. Please fix like this https://github.com/DroidKaigi/conference-app-2020/pull/597#discussion_r371398392 🙏

Reply:
OK!
I'll change the code.